### PR TITLE
Bans

### LINKED ISSUE #4602: set ban to expire after 1 days
in the admin page for bans, i modified the ban to expire after 1 day, i clicked on the modify button. i went back to main ban list where it shows the users ban will expire after 1 day(s), i click on modify and it shows that the ban will expire after 0 days. this is a bit confusing as i would expect it would say 1 day as i just set it to 1 day a few minutes ago. 0 days means it should expire today but the ban should expire tomorrow.
if you save/click modify with the number of days now set to 0 it shows on the ban list that the ban has now expired when it should still exist for less then 24 hours. if an admin goes back to modify a 1 day ban and does not see that the ban has gone to 0 days they will now have an expired ban rather then 1 day.

### PR DESCRIPTION
**Pull request by MissAllSunday**
fixes #4602 and #4601

It also adds a check on the "remove selected" buttons to not show the confirmation message if the user hasn't selected any ban to remove. Adds a generic txt string.

### CODE REVIEW

**Comment by jdarwood007** [comment]
If you don't select any ban and click remove selected, you get redirected back and the url shows bg=
The ban group info is lost.

A bug:

1. Set ban to 3 days and click modify
2. Return to the main ban list
3. Click Modify on the bang group
4. Ban days shows 2 now.
5. Repeat steps 2 and 3, ban is now 1 day.

**Comment by jdarwood007** [comment]
Does this also fix the bug I found where the time keeps decreasing?

**Comment by MissAllSunday** [comment]
no, I'm still working on that

**Comment by MissAllSunday** [comment]
So, the 3 then 2 then 1 day bug. It is actually doing what is been told, if you set a ban for 3 days at 10 am and then check again at 10:01am it will be no longer a 3 day ban, it will be a 2. something value which floor() will lower to 2.

ceil() would be the way to go here.

**Comment by jdarwood007** [comment]
As annoying as it will be.  Maybe we should go away from the simple "1" day ban.  Maybe a full date/time field/picker?

**Comment by MissAllSunday** [comment]
Nah, ceil will round up to the original value regardless of what time the ban was set, the one day stuff got covered on a previous commit.

**Comment by jdarwood007** [comment]
I realize that the first part of the bug I reported on was not on the ban list, but on the ban triggers page where the issue with remove selected occurs.

If you want to fix that let me know, otherwise I will merge this.

**Comment by MissAllSunday** [comment]
Its a consequence of using floor() on the edit page, it got lowered to 2 so 2 was saved and thats what it was showed on the ban list page.

BanList() is already using ceil() so fixing the modify ban page should cover the BanList() value too.

**Comment by MissAllSunday** [comment]
And I just realized you mean this page:

image


Thats where all the bugs are happening right?

**Comment by jdarwood007** [comment]
Yes that is correct.  The ceil did fix the first issue.  The remove selected issue is present on that page.  But since its not in the scope of t his PR, we can ignore it and report it separately or fix it with this PR.

**Comment by MissAllSunday** [comment]
OK that commit adds a confirmation message on the edit ban page, now both pages, ban list and ban edit has it.

**Comment by MissAllSunday** [comment]
Sorry I didn't understand you, what exactly did you mean @wintstar ?

File: Sources/ManageBans.php [hunk 1]
```diff
@@ -472,7 +486,9 @@ function BanEdit()
 					array(
 						'position' => 'below_table_data',
 						'value' => '
-						<input type="submit" name="remove_selection" value="' . $txt['ban_remove_selected_triggers'] . '" class="button"> <a class="button" href="' . $scripturl . '?action=admin;area=ban;sa=edittrigger;bg=' . $ban_group_id . '">' . $txt['ban_add_trigger'] . '</a>',
+						<input type="submit" name="remove_selection" value="' . $txt['ban_remove_selected_triggers'] . '" class="button"> <a class="button" 
+						href="' .
+							$scripturl	. '?action=admin;area=ban;sa=edittrigger;bg=' . $ban_group_id . '">' . $txt['ban_add_trigger'] . '</a>',
```
Comment (jdarwood007) [inline Sources/ManageBans.php:491]: I think this line here is being referenced.

**Comment by MissAllSunday** [comment]
Yep but why? Unless he means giving the html some nice formatting or something 



Jessica González, missallsunday.comEl Jeremy D <notifications@github.com>, 17 jul. 2018 19:55 escribió:@jdarwood007 commented on this pull request.



In Sources/ManageBans.php:
> @@ -472,7 +486,9 @@ function BanEdit()
 					array(
 						'position' => 'below_table_data',
 						'value' => '
-						<input type="submit" name="remove_selection" value="' . $txt['ban_remove_selected_triggers'] . '" class="button"> <a class="button" href="' . $scripturl . '?action=admin;area=ban;sa=edittrigger;bg=' . $ban_group_id . '">' . $txt['ban_add_trigger'] . '</a>',
+						<input type="submit" name="remove_selection" value="' . $txt['ban_remove_selected_triggers'] . '" class="button"> <a class="button" 
+						href="' .
+							$scripturl	. '?action=admin;area=ban;sa=edittrigger;bg=' . $ban_group_id . '">' . $txt['ban_add_trigger'] . '</a>',

I think this line here is being referenced.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/SimpleMachines/SMF2.1","title":"SimpleMachines/SMF2.1","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/SimpleMachines/SMF2.1"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jdarwood007 commented on #4825"}],"action":{"name":"View Pull Request","url":"https://github.com/SimpleMachines/SMF2.1/pull/4825#pullrequestreview-138071584"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/SimpleMachines/SMF2.1/pull/4825#pullrequestreview-138071584",
"url": "https://github.com/SimpleMachines/SMF2.1/pull/4825#pullrequestreview-138071584",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "@jdarwood007 commented on 4825",
"sections": [
{
"text": "",
"activityTitle": "**Jeremy D**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@jdarwood007",
"facts": [

]
}
],
"potentialAction": [
{
"targets": [
{
"os": "default",
"uri": "https://github.com/SimpleMachines/SMF2.1/pull/4825#pullrequestreview-138071584"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 354126119\n}"
}
],
"themeColor": "26292E"
}
]

**Pull request merged**